Neos.Flow/Classes/Security/AccountRepository.php

```diff
@@ -57,9 +58,9 @@ class AccountRepository extends Repository
     public function remove($object)
     {
         parent::remove($object);
-        /** @var Account $object */
-        $tag = 'Neos-Flow-Security-Account-' . md5($object->getAccountIdentifier());
-        $this->sessionManager->destroySessionsByTag($tag, sprintf('The account %s (%s) was deleted', $object->getAccountIdentifier(), $object->getAuthenticationProviderName()));
+        
+        // destroy the sessions for the account to be removed
+        (new SecurityContext)->destroySessionsForAccount($object);
```

Comment:
Not sure about this `(new SecurityContext)->` idiom, we use this nowhere and this also makes testing this method harder. I'd go with dependency injection. Also, the "reason" string is missing now `sprintf('The account %s (%s) was deleted', $object->getAccountIdentifier(), $object->getAuthenticationProviderName())`

Reply:
Many thanks for taking time to comment on this. I'm also unsure about this idiom; from my point-of-view it highlights a perceived architectural issue: dependency injection would inject _the_ security context of the current request, which is never needed; rather, _any_ security context will do, and is only needed in this one method.

What do you think: am I just "seeing things" or is this actually somewhat fine?

Reply:
You're right, this hints at an architectural smell. SecurityContext is specific to one session, but those methods are generic. This can of course be solved by introducing another layer of abstraction, but is that worth the added conceptual complexity?

Reply:
IMHO, this PR has already fixed one architectural smell: If you need to destroy sessions for accounts, you need to know exactly how to do that. So it is replacing one architectural smell for another, which is not necessarily bad.

Right now, the architecture in the bigger picture has no concept of a "remote session"; any notion of relations between Account and Session objects are strictly request-local and handled by the Security Context internally. I think something like a derived "Remote Security Context" might be what I'm looking for - but architecture on this scale is way beyond my skills for now.

Regarding worth, in the context of Neos as a product, there might be one if the Backend actually makes use of this (e.g. "User XXX is currently logged in" or "User XXX just updated a node with path /foo/bar") but right now, I don't see any good examples. In the context of Flow, it would be possible to implement functionality like "You are currently logged in on the following devices: …" with a remote-logout option (for example like WhatsApp Web, or Netflix).

In the context of #1051 there is no need to complicate the situation further.

Reply:
Oh I do see the use of this feature within Flow. I rather questioned if adding another LoA just to make the solution more architectural clean is worth it.

I'd probably go with injecting the SecurityContext and adding a documentation block, that explains why it's done in this place and that it's not for the current Context, but because the Context is the best place to hold the necessary knowledge to destroy arbitrary sessions by account.